### PR TITLE
Setting hostname to VM_NAME

### PR DESCRIPTION
**Pull request by nixjdm**
Also, this fixes provisioning with a repo branch specified. That previously failed to call the highstate.

### CODE REVIEW

File: salt_resources/states/top.sls [hunk 1]
```diff
@@ -4,6 +4,7 @@ base:
     - clean
     - basebox
     - basebox.symlink
+    - network
```
Comment (nixjdm) [inline salt_resources/states/top.sls]: I think we should do this all the time, thus it's uncommented.

File: Vagrantfile [hunk 2]
```diff
@@ -50,10 +50,15 @@ PROVISION_WITH_SALT = true
...
+
+# Set hostname + run highstate.
+PROVISION_CMD = SET_HOSTNAME + "sudo SSH_AUTH_SOCK=$SSH_AUTH_SOCK salt-call --local state.highstate --log-level info"
+
+# Add website branch var if applicable.
 if ENV["WEBSITE_BRANCH"]
-  PROVISION_CMD = "sudo SSH_AUTH_SOCK=$SSH_AUTH_SOCK salt-call --local grains.setval WEBSITE_BRANCH " + ENV["WEBSITE_BRANCH"] + " --log-level info"
-else
-  PROVISION_CMD = "sudo SSH_AUTH_SOCK=$SSH_AUTH_SOCK salt-call --local state.highstate --log-level info"
+  PROVISION_CMD = "sudo SSH_AUTH_SOCK=$SSH_AUTH_SOCK salt-call --local grains.setval Key Val --log-level info; " + PROVISION_CMD
```
Comment (verhulstm) [inline Vagrantfile]: this looks wrong
won't "grains.setval Key Val" actually set the key "key" and the value "val"?

File: salt_resources/states/network/init.sls [hunk 3]
```diff
@@ -0,0 +1,6 @@
+system:
+  network.system:
```
Comment (verhulstm) [inline salt_resources/states/network/init.sls]: deep system interaction like this are somewhat unstable in saltstack. it frustratingly enough often necessary to stick to simple sls states, like cli calls and file editing.

we can try using the network.system

File: Vagrantfile (hunk 2 above)
Comment (nixjdm) [inline Vagrantfile]: Whoops! Yeah, that's a mistake. Remnant of me debugging why the previous code didn't work. I'll fix it.

**Comment by nixjdm** [comment]
ref #48

**Pull request merged**